tooling/metamodel-generator/src/main/java/org/hibernate/processor/ClassWriter.java

```diff
@@ -238,7 +238,7 @@ else if ( modifiers.contains( Modifier.PROTECTED ) ) {
 		}
 		if ( entity.isImplementation() ) {
 			pw.print( entity.getElement().getKind() == ElementKind.CLASS ? " extends " : " implements " );
-			pw.print( entity.getSimpleName() );
+			pw.print( entity.importType( (entity.getQualifiedName()) ) );
```

Comment:
@cigaly This is probably OK, but I would prefer we added the import registration earlier, when we first recognize the inner interface, rather than doing it from `ClassWriter`. Is that hard to arrange?

Reply:
Most likely not big problem. I'll check that.

Reply:
@gavinking I've moved import from `ClassWriter` inito `AnnotationMetaEntity` constructor. I can not find other place to add it.

Additionally, I've replaced `replace('$', '.')` with `StringUtil.removeDollar` on two places in `ImportContextImpl` to avoid problem with importing classes which name ending in dollar sign. [Like `BookAuthorRepository$` in `NamedQueryTest`] If I remember correctly, this replacement is what `removeDollar` should do, anyway

Reply:
> I've moved import from ClassWriter inito AnnotationMetaEntity constructor.

Great, thanks.

> Additionally...

See my other comment, below.